### PR TITLE
Ako-yaswanth/ Include security teams pre commit hooks

### PR DESCRIPTION
**Pull request by ali-hosseini-deriv**
Changes:
add security team's secret scanner and run it through our husky pre-commit hook
-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: .husky/secret-scanner.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+if git log -1 > /dev/null 2>&1; then
+    $trufflehog_path git file://. --no-update --since-commit HEAD --fail > /tmp/trufflehog_output_$(whoami) 2>&1
+    trufflehog_exit_code=$?
+else
+    $trufflehog_path filesystem . --no-update --fail > /tmp/trufflehog_output_$(whoami) 2>&1
+    trufflehog_exit_code=$?
+fi
+
+# Only display results to stdout if trufflehog found something.
+if [ $trufflehog_exit_code -eq 183 ]; then
```
Comment (habib-deriv) [inline .husky/secret-scanner.sh:25]: Can we have a constant, what is this 183 representing?

File: .husky/secret-scanner.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    echo "Trufflehog is not setup. Please connect with Product Security Team"
+    exit 1
+fi
+
+# Use `filesysytem` if the git repo does not have any commits i.e its a new git repo.
+if git log -1 > /dev/null 2>&1; then
+    $trufflehog_path git file://. --no-update --since-commit HEAD --fail > /tmp/trufflehog_output_$(whoami) 2>&1
+    trufflehog_exit_code=$?
+else
+    $trufflehog_path filesystem . --no-update --fail > /tmp/trufflehog_output_$(whoami) 2>&1
```
Comment (habib-deriv) [inline .husky/secret-scanner.sh:20]: Let's store the $(whoami) in a variable and reuse it
const current_user=$(whoami)

**Review by shayan-deriv — APPROVED** [review]

File: .husky/secret-scanner.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+elif [[ -x /opt/homebrew/bin/trufflehog ]]; then
+    trufflehog_path="/opt/homebrew/bin/trufflehog"
+else
+    echo "Trufflehog is not setup. Please connect with Product Security Team"
+    exit 1
+fi
+
+# Use `filesysytem` if the git repo does not have any commits i.e its a new git repo.
+if git log -1 > /dev/null 2>&1; then
+    $trufflehog_path git file://. --no-update --since-commit HEAD --fail > /tmp/trufflehog_output_$(whoami) 2>&1
```
Comment (habib-deriv) [inline .husky/secret-scanner.sh:17]: /tmp/trufflehog_output_$(whoami) can be store in a variable as well

**Review by jim-deriv — APPROVED** [review]

File: .husky/secret-scanner.sh (hunk 3 above)
Comment (ritik-deriv) [reply .husky/secret-scanner.sh:17]: @habib-deriv The value of $(whoami) will be decided on runtime, we can store it in variable at the top but it looks redundant to me.

File: .husky/secret-scanner.sh (hunk 1 above)
Comment (ritik-deriv) [reply .husky/secret-scanner.sh:25]: This is the exit code of trufflehog which means that it found secrets.
Ref - https://github.com/trufflesecurity/trufflehog/blob/c51e8f8af5f4948ef92d5507a08e0228de0bdb63/README.md?plain=1#L251

**Review by habib-deriv — APPROVED** [review]

**Pull request closed without merging**